api/src/wfl/module/covid.clj

```diff
@@ -152,6 +152,22 @@
                              {:workload workload})))
     (if-not (or stopped finished) (stop! workload (utc-now)) workload)))
 
+(defn ^:private retry-covid-workload
+  "Retry/resubmit the `workflows` managed by the `workload` and return the
+   workload that manages the new workflows."
+  [{:keys [finished id executor] :as workload} workflows]
+  (when-not finished
+    (throw (UserException. "Cannot retry workload before it's finished."
+                           {:workload workload})))
+  ;; TODO: validate executor and sink.
```

Comment:
Seeking feedback:

Existing validation functionality is primed to validate a source/executor/sink request, not the resulting source/executor/sink itself.

I wasn't sure of the best Clojure-y way to take advantage of the existing validation functions, ex. for [TerraExecutor](https://github.com/broadinstitute/wfl/blob/develop/api/src/wfl/executor.clj#L446).  My idea is a multimethod to tag a stage with its `:name`, but I also looked at dispatching on multiple values (seems more elegant but I got confused).

Reply:
No I don't think I agree. I think one should be able to retry failed workflows while the workload is running.

Reply:
This decision is how I interpreted the [Spike doc](https://broadinstitute.atlassian.net/wiki/spaces/GHConfluence/pages/2674393095/Retries+in+Terra+WFL#Proposal):

"The /retry request … reanimates a workload if it was finished or stopped, …"

But the right interpretation may instead be to reanimate if needed, but not require the workload to be finished to retry.

The reason I put in this restriction is I was worried about overloading resources given the Rawls limitation on submitting snapshot subsets.  So resubmitting a snapshot with 500 resulting workflows, 1 failure and the rest running could result in concurrently running workflows for the same entity.  When we can submit a subset of a snapshot then I'd advise taking this restriction out.

Reply:
I think we should communicate this limitation rather than restrict people from doing what they need to do.

Reply:
I'll change this back (thinking of COVID processing, which is really one workload that never finishes unless we stop it for a WDL change or similar!)

Reply:
And re: validation (OG comment in thread), I'll file a ticket as validation in general could use further attention.

Specifically we do more than validate a request in this method, we also acquire resources as part of generating the stage object.  Maybe we will split out the validation and resource acquisition to be able to use validation as part of workload creation and retry.

Reply:
Validation ticket: https://broadinstitute.atlassian.net/browse/GH-1421